File: python/백준/재귀/25501 재귀의 귀재.py
def recursion(s, l, r, cnt):
    if l >= r: 
        return [1, cnt]
    elif s[l] != s[r]: 
        return [0, cnt]
    else: 
        return recursion(s, l+1, r-1, cnt + 1)

def isPalindrome(s, cnt):
    return recursion(s, 0, len(s) - 1, cnt)

T = int(input())

for _ in range(T):
    s = input().strip()  # 입력 문자열
    result = isPalindrome(s, 1)  # 회문 여부와 재귀 호출 횟수 반환
    print(" ".join(map(str, result)))  # 리스트를 문자열로 변환하여 출력

File: cpp/백준/재귀/25501 재귀의 귀재.cpp
#include <iostream>
#include <string>
#include <array>
using namespace std;

array<int, 2> recursion(const string& s, int l, int r, int cnt) {
    if (l >= r) return {1, cnt};
    if (s[l] != s[r]) return {0, cnt};
    return recursion(s, l + 1, r - 1, cnt + 1);
}

array<int, 2> isPalindrome(const string& s, int cnt) {
    return recursion(s, 0, (int)s.size() - 1, cnt);
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;

    while (T--) {
        string s;
        cin >> s;
        auto result = isPalindrome(s, 1);
        cout << result[0] << ' ' << result[1] << "\n";
    }
    return 0;
}
